### DISCUSSION TITLE
To add: Job::dispatch and __middleware with WithoutOverlapping and ->dontRelease() return that job is dispatched or rejected

### DISCUSSION THREAD

**Discussion by LocalHeroPro** [laravel/framework#38230] [Ideas]
Scenario:
```php
// in controller // dispatch
$a = JobName::dispatch();

// in JobName.php
public function middleware(): array
{
    return [(new WithoutOverlapping('awesome-name-' . $this->model->id)->dontRelease()];
}

// in controller
echo $a->isReleased(); // to handle if job is dispatched or not.
```

Right now we don't know if job is released or not.

**Comment by Cellard** [comment]
We had very hard debug with the same scenario.

I don't know how and why, but atom lock flag was seized up in Redis (maybe the queue system were died while working with a job). So every next job was not even handled, being stopped by middleware. But these jobs were marked as `completed`, so we can't understand what's going on for a long time.

First, I strictly recommend not to run `WithoutOverlapping` middleware with default `expiresAfter` value.
Second, I think that rejected job should not be marked as `completed`. I think such a job should be marked as `failed` or something like that.

Maybe that job should throw a proper exception, so we can examine it in `failed_jobs` table or horizon dashboard.
